src/main/kotlin/graphics/scenery/backends/vulkan/VulkanCommandBuffer.kt

```diff
@@ -1,70 +1,48 @@
 package graphics.scenery.backends.vulkan
 
 import graphics.scenery.utils.LazyLogger
-import org.lwjgl.vulkan.*
-import org.lwjgl.vulkan.VK10.*
-import org.lwjgl.system.MemoryUtil.*
-import java.nio.LongBuffer
+import org.lwjgl.system.MemoryUtil.NULL
+import org.lwjgl.vulkan.VkCommandBuffer
+import vkn.*
 
 /**
  * Vulkan Command Buffer class. Wraps command buffer and fencing functionality.
  *
  * @author Ulrik Günther <hello@ulrik.is>
  */
-class VulkanCommandBuffer(val device: VulkanDevice, var commandBuffer: VkCommandBuffer?, val fenced: Boolean = true): AutoCloseable {
+class VulkanCommandBuffer(val device: VulkanDevice, var commandBuffer: VkCommandBuffer?, val fenced: Boolean = true) : AutoCloseable {
     private val logger by LazyLogger()
     var stale: Boolean = true
 
     private var fenceInitialized: Boolean = false
-    private var fence: LongBuffer = memAllocLong(1)
+    private var fence: VkFence = NULL
     var submitted = false
 
     init {
-        if(fenced) {
+        if (fenced)
             addFence()
-        }
     }
 
     fun addFence() {
-        val fc = VkFenceCreateInfo.calloc()
-            .sType(VK_STRUCTURE_TYPE_FENCE_CREATE_INFO)
-            .pNext(NULL)
-
-        val f = VU.getLong("Creating fence",
-            { vkCreateFence(device.vulkanDevice, fc, null, this) },
-            { fc.free() })
-
-        fence.put(0, f)
+        fence = device.vulkanDevice createFence vk.FenceCreateInfo { }
```

Comment:
The use of infix functions in these cases looks unintuitive to me. Would extension functions be an alternative?

Reply:
you mean 

`device.vulkanDevice.createFence(vk.FenceCreateInfo { })`

?